VectorWithOffset constructor warnings

https://github.com/UCL/STIR/blob/edccb82c089a34789544f7c2e83241927dd54d63/src/include/stir/VectorWithOffset.inl#L279-L280
gives warnings as the wrong order of arguments is used
```
/home/kris/devel/build/SIRF-SuperBuild/builds/STIR/build/src/swig/CMakeFiles/_stir.dir/stirPYTHON_wrap.cxx:48419:106:   required from here
/home/kris/devel/build/SIRF-SuperBuild/sources/STIR/src/include/stir/VectorWithOffset.h:400:16: warning: ‘stir::VectorWithOffset<float>::pointer_access’ will be initialized after [-Wreorder]
  400 |   mutable bool pointer_access;
      |                ^~~~~~~~~~~~~~
/home/kris/devel/build/SIRF-SuperBuild/sources/STIR/src/include/stir/VectorWithOffset.h:390:19: warning:   ‘std::shared_ptr<float []> stir::VectorWithOffset<float>::allocated_memory_sptr’ [-Wreorder]
  390 |   shared_ptr<T[]> allocated_memory_sptr;
      |                   ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/kris/devel/build/SIRF-SuperBuild/sources/STIR/src/include/stir/VectorWithOffset.h:405,
                 from /home/kris/devel/build/SIRF-SuperBuild/sources/STIR/src/include/stir/stream.h:28,
```

--- a/src/include/stir/VectorWithOffset.inl
+++ b/src/include/stir/VectorWithOffset.inl
@@ -276,8 +276,8 @@ template <class T>
 VectorWithOffset<T>::VectorWithOffset(const int min_index, const int max_index, T* const data_ptr, T* const end_of_data_ptr)
     : length(static_cast<unsigned>(max_index - min_index) + 1),
       start(min_index),
-      pointer_access(false),
-      allocated_memory_sptr(nullptr) // we don't own the data
+      allocated_memory_sptr(nullptr), // we don't own the data
+      pointer_access(false)      
 {
   this->begin_allocated_memory = data_ptr;
   this->end_allocated_memory = end_of_data_ptr;

--- a/src/include/stir/VectorWithOffset.inl
+++ b/src/include/stir/VectorWithOffset.inl
@@ -277,7 +277,7 @@ VectorWithOffset<T>::VectorWithOffset(const int min_index, const int max_index, 
     : length(static_cast<unsigned>(max_index - min_index) + 1),
       start(min_index),
       allocated_memory_sptr(nullptr), // we don't own the data
-      pointer_access(false)      
+      pointer_access(false)
 {
   this->begin_allocated_memory = data_ptr;
   this->end_allocated_memory = end_of_data_ptr;

